src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -104,18 +105,40 @@ public String getPagedResponseWrappersClassName() {
   }
 
   /** The name of the generated resource type from the entity name. */
-  public Name getResourceTypeName(String entityName) {
-    return Name.from(entityName).join("name");
+  public Name getResourceTypeName(String entityName, ResourceNameType resourceNameType) {
```

Comment:
It is used outside the surface namer here:
https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java#L363

But this is only for a local var name. And because it is in the Java test transformer, the upper camel format is known, so have updated as you suggest.